core/src/main/java/org/akaza/openclinica/service/crfdata/xform/EnketoAPI.java

```diff
@@ -164,21 +167,20 @@ private EnketoURLResponse registerAndGetURL(URL url, String crfOID) {
         }
     }
 
-
     public EnketoURLResponse registerAndGetEditURL(FormLayout formLayout, String crfFlavor, String instance, String ecid, String redirect, boolean markComplete,
-            String studyOid, List<FormLayoutMedia> mediaList, String goTo, String flavor, Role role, Study parentStudy, StudyEvent studyEvent, String mode) {
+            String studyOid, List<FormLayoutMedia> mediaList, String goTo, String flavor, Role role, Study parentStudy, StudyEvent studyEvent, String mode,
+            EventDefinitionCrf edc) {
         EnketoURLResponse urlResponse = null;
         try {
-            //  Role role, Study parentStudy, StudyEvent studyEvent, String mode
-            urlResponse = getEditURL(formLayout, crfFlavor, instance, ecid, redirect, markComplete, studyOid, mediaList, goTo, flavor,
-                    role, parentStudy, studyEvent, mode);
+            // Role role, Study parentStudy, StudyEvent studyEvent, String mode
+            urlResponse = getEditURL(formLayout, crfFlavor, instance, ecid, redirect, markComplete, studyOid, mediaList, goTo, flavor, role, parentStudy,
+                    studyEvent, mode, edc);
         } catch (Exception e) {
-            if (StringUtils.equalsIgnoreCase(e.getMessage(), "401 Unauthorized")
-                    || StringUtils.equalsIgnoreCase(e.getMessage(), "403 Forbidden")) {
+            if (StringUtils.equalsIgnoreCase(e.getMessage(), "401 Unauthorized") || StringUtils.equalsIgnoreCase(e.getMessage(), "403 Forbidden")) {
                 savePformRegistration();
                 try {
-                    urlResponse = getEditURL(formLayout, crfFlavor, instance, ecid, redirect, markComplete, studyOid, mediaList, goTo, flavor,
-                            role, parentStudy, studyEvent, mode);
+                    urlResponse = getEditURL(formLayout, crfFlavor, instance, ecid, redirect, markComplete, studyOid, mediaList, goTo, flavor, role,
```

Comment:
I am going to merge this but we really need to think about how to avoid doing this...there are 15 parameters to this method now. We keep on adding more. Let me know if you've any ideas.